### PR TITLE
Add number range search options

### PR DESCRIPTION
**Pull request by doekenorg**
This PR addresses #1426 by adding a new `number_range` field for `number`, `quantity` and `total` fields. For `products` this isn't possible as the field itself contains both a name AND a price.

This PR also resolves an issue with the removal of `CAST` functions from the query, that we actually still need in order to match numeric values. 

image

min-max

There is some JS that prevents the From field from being higher than the value of the To field, and vice versa. There is also a backup validation with an error message if someone gets around it.

### CODE REVIEW

**Comment by doekenorg** [comment]
- [ ] Needs a test case

**Comment by mrcasual** [comment]
@doekenorg, please take a look at this: https://www.loom.com/share/96864d158a8b46dc987a86e4f361c7bb

**Comment by mrcasual** [comment]
@doekenorg, also, are you able to add support for the product (user-defined price) field?

File: includes/widgets/search-widget/templates/search-field-number_range.php [hunk 1]
```diff
@@ -19,21 +19,41 @@
...
+$is_currency = 'total' === $gravityview_view->search_field['type'];
+if ( ! $is_currency ) {
+	// could still be currency from the field.
+	$field       = GVCommon::get_field( $gravityview_view->getForm() ?? [], $gravityview_view->search_field['key'] );
+	$is_currency = 'currency' === $field->numberFormat;
+}
+
 // Add decimal support for total price.
 $step = apply_filters(
 	'gk/gravityview/search/number_range/step',
```
Comment (mrcasual) [inline includes/widgets/search-widget/templates/search-field-number_range.php:31]: Please add docblock for the filter. Also, `gk/gravityview/foo-bar/baz`.

**Pull request merged**